### PR TITLE
Update JDC copy to match governance documents

### PR DESCRIPTION
**Pull request by afshin**
The overview section of the governance docs has the relevant text, so I have just used the same copy here for consistency.

https://github.com/jupyter/governance/blob/8ea06d0d1620fcd8154129e69b2e93c2fa698adc/overview.md?plain=1#L22

### CODE REVIEW

File: about.html [hunk 1]
```diff
@@ -16,7 +16,7 @@
...
 
 contributors_copy: |
-  Project Jupyter <a href="https://jupyter.org/governance/distinguished_contributors.html">Distinguished Contributors</a> are recognized for their substantial contributions to Jupyter itself in both quality and quantity over at least two years. Contributions may include code, code review, infrastructure work, mailing list and chat participation, community help/building, education and outreach, fundraising, branding, marketing, inclusion and diversity, UX design and research, etc. Up to 10 new Distinguished Contributors are selected each year by the cumulative body of Distinguished Contributors.
+  Project Jupyter <a href="https://jupyter.org/governance/distinguished_contributors.html">Distinguished Contributors</a> are a group of Jupyter community members that have gone above-and-beyond in their support of the project over the years, making substantial and sustained contributions in any area of activity (software development, governance, community engagement, events, etc.). The Jupyter community confers membership in this group as a way of recognizing their effort and saying "thank you."
```
Comment (JasonWeill) [inline about.html:19]:
```suggestion
  Project Jupyter <a href="https://jupyter.org/governance/distinguished_contributors.html">Distinguished Contributors</a> are a group of Jupyter community members that have gone above and beyond in their support of the project over the years, making substantial and sustained contributions in any area of activity, such as software development, governance, community engagement, and events. The Jupyter community confers membership in this group as a way of recognizing and thanking members for their effort.
```

Minor copy edit (should probably be made in the linked source material as well)

**Review by jasongrout — APPROVED** [review]
LGTM, especially after @JasonWeill's edits.

File: about.html (hunk 1 above)
Comment (afshin) [reply about.html:19]: Please let's make the edit in the source text first because the whole point of this PR is to rationalize the two documents.

**Comment by jasongrout** [comment]
Now that we've opened an issue to adjust the source wording, I'll merge this. We can then copy any new source wording back to here. Thanks again @JasonWeill - I think your text reads much better.

**Pull request merged**